server/user/user.controller.js

```diff
@@ -25,6 +27,33 @@ function get(req, res) {
   return res.json(req.user);
 }
 
+function login(req, res, next) {
+  User.findOne({ username: req.body.username }, (error, doc) => {
+    if (!doc) {
+      const error1 = new APIError(`Error: Account with username not found! ${req.body.username}`, httpStatus.UNAUTHORIZED);
+      next(error1);
+      return;
+    }
+    bcrypt.compare(req.body.password, doc.password, (err, result) => {
+      if (err) {
```

Comment:
instead of doing the same thing below we can change this to `if (err || !result) {`